### PR TITLE
Use copies of the estimates when shuffling their elements instead of doing it in-place

### PR DESCRIPTION
**Pull request by shak360**
Python error where I thought creating a temporary copy of the T matrix would create a different T to alter, but it was actually changing the original T. 

Now uses `np.copy()`.

### CODE REVIEW

File: lineage/Analyze.py [hunk 1]
```diff
@@ -164,33 +163,42 @@ def Results(tHMMobj, pred_states_by_lineage, LL):
...
-    results_dict["transition_matrix_norm"] = np.linalg.norm(temp_T - tHMMobj.X[0].T)
-
+    if fT is None:
+        results_dict["transition_matrix_norm"] = np.linalg.norm(temp_T - tHMMobj.X[0].T)
+    else:
+        results_dict["transition_matrix_norm"] = np.linalg.norm(tHMMobj.estimate.T - tHMMobj.X[0].T)
+        
     # Rearrange the values in the pi vector
-    temp_pi = tHMMobj.estimate.pi
+    temp_pi = np.zeros((tHMMobj.num_states))
```
Comment (shak360) [inline lineage/Analyze.py:178]: This line is one of the issues where copying the estimate vector and then changing the copy also changed the original (which isn't what we want).

File: lineage/Analyze.py [hunk 2]
```diff
@@ -164,33 +163,42 @@ def Results(tHMMobj, pred_states_by_lineage, LL):
     results_dict["switcher_map"] = switcher_map
 
     # Rearrange the values in the transition matrix
-    temp_T = tHMMobj.estimate.T
+    temp_T = np.zeros((tHMMobj.num_states,tHMMobj.num_states))
```
Comment (shak360) [inline lineage/Analyze.py:166]: The other time where I copy the estimate T and then change the copy, but this also changes the original. Now it uses a matrix that is no where related to the original copy, so changing it doesn't affect the original.

**Comment by shak360** [comment]
So on `master`, if we try to make a figure when fixing T and pi, we get the following:

image

Note how the errors for the T matrix norm and the pi vector norm _should_ be zero (because they're fixed) but aren't. This is because of the errors above. After this fix, on this branch this figure looks like the following:

image

File: lineage/Analyze.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def run_Analyze_over(list_of_populations, num_states, parallel=True, **kwargs):
     return output
 
 
-def Results(tHMMobj, pred_states_by_lineage, LL):
+def Results(tHMMobj, pred_states_by_lineage, LL, fpi, fT, fE):
```
Comment (shak360) [inline lineage/Analyze.py:99]: The Results function should know if we have fixed parameters that were given in analysis.

File: lineage/Analyze.py [hunk 4]
```diff
@@ -221,15 +229,15 @@ def Results(tHMMobj, pred_states_by_lineage, LL):
     return results_dict
 
 
-def run_Results_over(output):
+def run_Results_over(output, list_of_fpi, list_of_fT, list_of_fE):
```
Comment (shak360) [inline lineage/Analyze.py:232]: The 'Results' function should know if we have fixed parameters.

File: lineage/Analyze.py (hunk 3 above)
Comment (shak360) [reply lineage/Analyze.py]: Outdated, ignore.

File: lineage/Analyze.py (hunk 4 above)
Comment (shak360) [reply lineage/Analyze.py]: Outdated, ignore.

File: lineage/Analyze.py (hunk 1 above)
Comment (aarmey) [reply lineage/Analyze.py:174]: You want `np.copy()`

https://numpy.org/doc/1.18/reference/generated/numpy.copy.html

**Pull request merged**